### PR TITLE
Alerting: Introduces `/api/v1/ngalert/alertmanagers` to expose discovered and dropped Alertmanager(s)

### PR DESCRIPTION
**Pull request by gotjosh**
Exposing for visibility but this description will get amended.

### CODE REVIEW

**Review by papagian — COMMENTED** [review]
I have tested and it works fine!
```
###
# set external Alertmanager
POST http://admin:admin@localhost:3000/api/v1/ngalert/admin_config
content-type: application/json

{
  "alertmanagers": ["http://localhost:9093"]
}

###
GET http://admin:admin@localhost:3000/api/v1/ngalert/admin_config

###
# after a few minutes it should be discovered
GET http://admin:admin@localhost:3000/api/v1/ngalert/alertmanagers

###
# remove it
POST http://admin:admin@localhost:3000/api/v1/ngalert/admin_config
content-type: application/json

{
  "alertmanagers": []
}

###
# check again
GET http://admin:admin@localhost:3000/api/v1/ngalert/alertmanagers
```

The only issue is the generated API [spec](https://editor.swagger.io/?url=https://raw.githubusercontent.com/grafana/grafana/api-discovered-alertmanagers/pkg/services/ngalert/api/tooling/post.json) where some responses (eg `GettableAlerts`) are empty.

File: docs/sources/administration/configuration.md [hunk 1]
```diff
@@ -1097,6 +1097,16 @@ Sets a global limit on number of users that can be logged in at one time. Defaul
...
 Sets a global limit on number of alert rules that can be created. Default is -1 (unlimited).
 
+<hr>
+
+## [ngalerting]
+For more information about the NGAlert feature in Grafana, refer to [Unified Alerting]({{< relref "../unified-alerting/_index.md" >}}}).
+
+### admin_config_poll_interval_seconds
+
+How frequently we poll for admin config changes. Default value is `60`.
```
Comment (achatterjee-grafana) [inline docs/sources/administration/configuration.md:1107]:
```suggestion
 Specify the frequency of polling for admin config changes. The default value is `60`.
```

**Review by achatterjee-grafana — COMMENTED** [review]
Added doc review.

File: docs/sources/administration/configuration.md [hunk 2]
```diff
@@ -1097,6 +1097,16 @@ Sets a global limit on number of users that can be logged in at one time. Defaul
 
 Sets a global limit on number of alert rules that can be created. Default is -1 (unlimited).
 
+<hr>
+
+## [ngalerting]
```
[the remark this answers is not in the document]
Comment (gotjosh) [reply docs/sources/administration/configuration.md:1102]: @torkelo @kylebrandt any thoughts?

File: docs/sources/administration/configuration.md (hunk 1 above)
Comment (gotjosh) [reply docs/sources/administration/configuration.md:1107]: Thanks, will update.

File: docs/sources/administration/configuration.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (gotjosh) [reply docs/sources/administration/configuration.md]: we have settled on `unified-alerting`.

**Review by papagian — APPROVED** [review]

**Review by achatterjee-grafana — APPROVED** [review]
Doc content looks good.

**Pull request merged**